packages/dds/tree/src/shared-tree/independentView.ts

```diff
@@ -117,21 +111,55 @@ export function independentInitializedView<const TSchema extends ImplicitFieldSc
 
 	const fieldCursors = fieldBatchCodec.decode(content.tree as JsonCompatibleReadOnly, context);
 	assert(fieldCursors.length === 1, 0xa5b /* must have exactly 1 field in batch */);
-	// Checked above.
-	// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-	initializeForest(forest, fieldCursors[0]!, revisionTagCodec, idCompressor, false);
+
+	const out: TreeViewAlpha<TSchema> = independentInitializedViewInternal<TSchema>(
+		config,
+		options,
+		schema,
+		// Checked above.
+		// eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+		fieldCursors[0]!,
+		idCompressor,
+	);
+	return out;
+}
+
+/**
+ * {@link independentInitializedView} but using internal types instead of persisted data formats.
+ */
+export function independentInitializedViewInternal<const TSchema extends ImplicitFieldSchema>(
```

Comment:
I have found that for some testing edge cases, where creating views with unknown optional fields in the stored schema and document content, having this function is handy, so I split it into the two parts.

The bug this PR fixes is solved by the new function using initialize on the checkout (which does the multi phase schema changes allowing the optional root before the root is written) instead of initializeForest.